src/test/bassTranslatorTest.js

```diff
@@ -0,0 +1,189 @@
+                            break
+                        }
+                    }
+                }
+            }
+        }
+
+        //Utils.log("alterationsInCurrentMeasure", JSON.stringify(alterationsInCurrentMeasure))
+        //Utils.log("toOmit", JSON.stringify(toOmit))
+        //Utils.log("toExtra", JSON.stringify(toExtra))
+
+        if (toOmit.length === 1 && contains(toOmit, 3)) {
+            if (mode === "minor" && toExtra[0] === ">3") {
+                harmonicFunctions[0][i].mode = "major";
+            }
+            if (mode === "major" && toExtra[0] === "<3") {
+                harmonicFunctions[0][i].mode = "minor";
+            }
+        } else {
+            for (var a = 0; a < toOmit.length; a++) {
+
+                if (harmonicFunctions[0][i].omit === undefined) {
+                    harmonicFunctions[0][i].omit = []
+                }
+                if (harmonicFunctions[0][i].extra === undefined) {
+                    harmonicFunctions[0][i].extra = []
+                }
+
+                harmonicFunctions[0][i].omit.push(toOmit[a])
+                harmonicFunctions[0][i].extra.push(toExtra[a])
+            }
+        }
+
+        summedNotesLength = addNotesLength(summedNotesLength, durations[i])
+
+        if (notesLengthEqual(summedNotesLength, meter)) {
+            alterationsInCurrentMeasure = []
+            summedNotesLength = [0, 0]
+        }
+    }
+}
+
+
+var harmonicFunctions = JSON.parse('[[{"functionName":"T","degree":1,"revolution":"1","extra":[],"omit":[],"down":false},{"functionName":"D","degree":5,"revolution":"1","extra":[],"omit":[],"down":false},{"functionName":"T","degree":1,"revolution":"5","extra":[],"omit":[],"down":false},{"functionName":"S","degree":4,"revolution":"3","extra":[],"omit":[],"down":false},{"functionName":"T","degree":1,"revolution":"1","extra":[],"omit":[],"down":false}]]')
+var chordElements = JSON.parse('[{"notesNumbers":[0,2,4],"omit":[],"bassElement":{"bassNote":{"pitch":48,"baseNote":0,"chordComponent":1},"symbols":[{"component":3},{"component":5}]},"primeNote":0},{"notesNumbers":[4,6,8],"omit":[],"bassElement":{"bassNote":{"pitch":43,"baseNote":4,"chordComponent":1},"symbols":[{"component":3},{"component":5}]},"primeNote":4},{"notesNumbers":[4,9,7],"omit":[],"bassElement":{"bassNote":{"pitch":43,"baseNote":4,"chordComponent":5},"symbols":[{"component":6,"alteration":"b"},{"component":4}]},"primeNote":0},{"notesNumbers":[5,10,7],"omit":[],"bassElement":{"bassNote":{"pitch":45,"baseNote":5,"chordComponent":3},"symbols":[{"component":6},{"component":3}]},"primeNote":3},{"notesNumbers":[0,2,4],"omit":[],"bassElement":{"bassNote":{"pitch":48,"baseNote":0,"chordComponent":1},"symbols":[{"component":3},{"component":5}]},"primeNote":0}]')
+var mode = "major"
+var meter = [4, 4]
+var durations = [[1, 2], [1, 2], [1, 2], [1, 4], [1, 4]]
+
+
+handleAlterations(harmonicFunctions, chordElements, mode, meter, durations)
+
+console.log(JSON.stringify(harmonicFunctions))
```

Comment:
There are available some testing tools - look up into other test classes how to use them and please try to change it.

Reply:
okay :D